storage/src/direct_kv_block.cpp

```diff
@@ -48,9 +49,9 @@ Sliver create(const SetOfKeyValuePairs &updates,
     header->numberOfElements = numOfElements;
     std::int32_t currentOffset = metadataSize;
     auto *entries = (detail::Entry *)(blockBuffer + sizeof(detail::Header));
-    for (const auto &elem : updates) {
-      const KeyValuePair &kvPair = elem;
-
+    // Serialize key/values in a deterministic order.
+    const auto orderedUpdates = order(updates);
```

Comment:
The `order()` function returns an object that is created from its argument. Copying will either be optimised away through return value optimisation (RVO) or it will be moved. I would prefer we leave as is, because otherwise we will be binding to a temporary and it might be a bit misleading as to what the intent is.